src/lib/common/include/sol-update.h

```diff
@@ -85,9 +85,9 @@ struct sol_update_info {
 #define SOL_UPDATE_INFO_API_VERSION (1)
     uint16_t api_version;
 #endif
+    bool need_update; /**< @brief If version of update is newer than current, so the update is necessary. */
```

Comment:
There's now a hole of alignof(version)-alignof(need_update) bytes in this struct. This is 7 bytes on x86-64, and 3 bytes on x86. You might want to [read this article on struct packing](http://www.catb.org/esr/structure-packing/), since the reorganization in all the structs in this patch is not doing what's in the tin (quite the contrary, actually).